server/http.go

```diff
@@ -639,6 +667,47 @@ func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (p *Plugin) handleChannelPreference(w http.ResponseWriter, r *http.Request) {
+	submitRequest := &model.SubmitDialogRequest{}
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&submitRequest); err != nil {
+		p.API.LogError("Error decoding dialog request", "Error", err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if submitRequest.UserId == "" {
+		p.API.LogError("Invalid user ID", "UserID", submitRequest.UserId)
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	if !p.API.HasPermissionTo(submitRequest.UserId, model.PermissionManageSystem) {
+		p.API.LogError("Unable to resolve request due to insufficient permissions", "UserID", submitRequest.UserId)
+		http.Error(w, "Insufficient permissions", http.StatusForbidden)
+		return
+	}
+
+	zoomChannelSettingsMapValue := ZoomChannelSettingsMapValue{
+		ChannelName: submitRequest.CallbackId,
```

Comment:
We're using the display name so probably best to put that here. A little unfortunate that this will become stale if the channel's display name changes. Maybe we should use `channel.Name` instead, which is generally more permanent? The user can change both values, but it's more likely that a display name gets changed. Storing it *with* the channel id below makes sense. I would expect the channel id to be in the struct as well, but it's obviously not required since the feature currently works without that

```suggestion
		ChannelDisplayName: submitRequest.CallbackId,
```